### ISSUE TITLE
Main.getClassList does not allow to include single picard tools

### ISSUE THREAD

**Issue by magicDGS** [broadinstitute/gatk#4660]
For downstream projects, there is a way to bundle single class files into the toolkit by adding them to the `Main.getClassList`; nevertheless, this only allows to include classes extending the `org.broadinstitute.hellbender.cmdline.CommandLineProgram`. For including a tool from picard, the only way is to add a whole package where `picard.cmdline.CommandLineProgram` extensions are located.

Either a new method for include single classes from Picard should be added, or https://github.com/broadinstitute/barclay/issues/127 implemented (a proposal has been done in https://github.com/broadinstitute/barclay/pull/96) and move Picard/GATK to use the common interface.

**Comment by magicDGS** [comment]
The option of a common class would be useful also to get rid of the decorator around picard tools (`PicardCommandLineProgramExecutor`)

**Comment by cmnbroad** [comment]
@magicDGS We already plan to add the common CommandLineProgram interface as described in (broadinstitute/barclay#127). I think that makes a lot more sense than adding more new methods for this. The existing PR for it is out of date though, since there is more and more common functionality that could be pushed up into the common base (beta/experimental decoration, tool registry, etc.).

**Comment by magicDGS** [comment]
I agree with the common functionality, but the PR aims to set the less constraints as possible for the custom implementations. Let me know here or in https://github.com/broadinstitute/barclay/issues/127 the functionality to be common in the base class, and I can work on an updated PR for it.

**Comment by magicDGS** [comment]
@cmnbroad - what is the timeframe for a common CLP class? Because in the meantime it would be nice to be able to add picard tools, and anyway the common CLP would require refactoring in `Main`. So is it too much problem if I add a temporary method to add single picard tools?
